_posts/2020-10-22-modele-wdrozenia-w-chmurze.md

```diff
@@ -0,0 +1,199 @@
+
+FaaS najlepiej sprawdza się w przetwarzaniu opartym o zdarzenia. Zastosowanie funkcji do tworzenia aplikacji webowych lub usług API należy poprzedzić analizą wymagań, ocenić np. dopuszczalne SLA usługi przy uwzględnieniu czasu zimnego startu. Możliwość skalowania usług od zera do dowolnego obciążenia oraz ograniczenie czasu tworzenia nowych funkcjonalności do minimum sprawia, że FaaS to zawsze opcja warta rozważenia.
+
+Niskie koszty utrzymania infrastruktury, wysokie tempo wprowadzania zmian i wzorowe możliwości skalowania pozwolą Twojemu produktowi zyskać przewagę nad konkurencją.
+
+# Migracja a rozwój
+
+Przytaczane do tej pory koszty związane z migracją zakładały dostosowanie i wdrożenie działających systemów. Przy ich ocenie zakładałem, że zespół nie posiada wiedzy i doświadczenia w tworzeniu aplikacji w chmurze. W takiej sytuacji koszt szkoleń, prób i błędów oraz faktycznego wdrożenia rośnie wraz ze wzrostem "chmurowości" rozwiązania.
+
+Sytuacja będzie wyglądać zupełnie inaczej dla zespołu tworzącego i utrzymującego już rozwiązania chmurowe. W takiej sytuacji koszt nowej technologii został już poniesiony i nie wpływa na koszt przygotowywania nowych rozwiązań. Dla doświadczonych zespołów koszt dostarczenia nowej funkcjonalności w środowisku chmurowym będzie wyraźnie niższy niż w tradycyjnym podejściu ze względu na:
+- abstrakcję infrastruktury,
+- minimalny koszt wdrożenia i utrzymania aplikacji,
+- przeniesienie ciężaru na rozwój funkcjonalny aplikacji,
+- dostępność gotowych do integracji z systemem usług w modelu SaaS
+
+# Vendor lock-in
+
+Planując migrację do chmury musisz zmierzyć się z tematem uzależnienia kodu od konkretnego dostawcy.
+
+Czy _vendor lock-in_ jest zły? Zagadnienie warto traktować jako ryzyko, a nie zagrożenie jako takie. Dla ryzyk możliwe jest zaplanowanie ścieżek przeciwdziałania i przygotowanie planu wyjścia z chmury (czy to do innego dostawcy, czy zupełnie w kierunku chmury prywatnej lub rozwiązania _on-premises_).
+
+Dla podejść IaaS i CaaS możemy przyjąć, że ryzyko wyjścia z chmury niesie za sobą minimalne koszty - w szczególności, gdy system nie korzysta z natywnych usług oferowanych przez chmurę obliczeniową.
+
+W przypadku PaaS i FaaS prawdopodobnie zaprojektowałeś system pod kątem oferty konkretnej chmury. Najwięksi dostawcy zapewniają ujednoliconą ofertę produktów. O ile różnice pomiędzy oferowanymi rozwiązaniami w wielu przypadkach uniemożliwiają migrację systemu bez żadnych zmian, to jednak znajdziemy pomiędzy nimi odpowiadające sobie rozwiązania pozwalające zrealizować system w podobnej architekturze.
+
+Warto zauważyć, że problem _vendor lock-in_, jako poważne ryzyko migracji do chmury, często jest adresowany wprost przez dostawców. Większość produktów Google Cloud Platform jest zbudowana w oparciu o otwarte standardy oraz narzędzia open-source. Przykładowo Google Kubernetes Engine do działania wykorzystuje platformę Kubernetes, Cloud Bigtable jest zgodny z HBase, itd.
```

Comment:
Jest to jedyny punkt w którym przywołane jest GCP. Może dałoby się to tak uogólnić, żeby cały artykuł był "vendor agnostic" Oczywiście można gdzieś (a nawet trzeba) dać w meta "azure,aws, gcp" żeby kliki były :D

Reply:
hmm, zastanawiałem się nad tym.
widzę dwie opcje
- tutaj dam magiczne "na przykład" z adnotacją że inni też się starają
- wplotę więcej przykładów nt. GCP w cały art

jak myślisz?

Reply:
zmienię kierunek na 'przykład', nie chcę już rozciągać wpisu :)